src/plotting/plotSettings.component.tsx

```diff
@@ -80,6 +81,94 @@ const StyledTab = styled(Tab)(() => ({
   width: 10,
 }));
 
+/**
+ * Handles the colours currently in use for plotting different channels
+ * Determines which colours we have available based on which have already been selected
+ * Uses a list of 10 pre-selected colours before generating a random colour beyond this
+ */
+export class ColourGenerator {
+  selectedColours: string[];
+  remainingColours: string[];
+
+  // List of colours to generate in order (taken from eCat)
```

Comment:
I was gonna choose a colour palette from ColorBrewer e.g. https://colorbrewer2.org/#type=qualitative&scheme=Set1&n=9 - I suppose we can't go wrong with picking what eCat had - they're just a tag ugly imo 😆

Reply:
My logic was that they can't possibly complain since these are the colours they currently use ;) but I am very open to a separate colour palette if you wanted me to. I haven't used that website before, would you like to recommend one? We'd have to consider what looks best on a plot with a white background (e.g. no grey or very light colours if possible)

Reply:
Let's just stick with the existing one - it's easy to change it later. The one I linked was the one I was thinking of (9 class set 1).

Reply:
An alternative for 10, in case we come back to this later - this seems to be a literal rainbow so could work quite nicely - https://colorbrewer2.org/#type=diverging&scheme=Spectral&n=10

Reply:
Yeah, my only thought against it is that the shades are quite similar because it's a "heat map" style colour scheme - which could be confusing